.github/workflows/publish.yml

```diff
@@ -0,0 +1,66 @@
+name: Publish artifacts
```

Comment:
I would actually suggest splitting this out into two separate workflows.

e.g.

- [`.github/workflows/docker.yml`](https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/.github/workflows/docker.yml)
- [`.github/workflows/publish-package.yml`](https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/.github/workflows/publish-package.yml)

It seems strange I know, as they are doing the same goal, but having workflows being very specific and factorized is the conclusion we came to after a year or so of maintaining the GHA CI system. Just a suggestion though.

Reply:
I understand that could be beneficial if one of the job specifications increase size drastically (i.e. pyhf's [docker.yml](https://github.com/scikit-hep/pyhf/blob/70f100c56588e4119eca5d00ddb101911a4cd50e/.github/workflows/docker.yml) file with 130 LOC), but given the simplicity of these jobs steps, and the **shared** trigger condition, I would leave them together.